### PR TITLE
fix: issues in exporter binary

### LINKED ISSUE #182: [BUG] Read Doc failed error when running the exporter with the binaries
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What version of the Terraform exporter are you using?

0.1.0-beta

### What version of the Terraform CLI are you using?

1.9.8

### What type of issue are you facing

bug report

### Describe the bug

When running the exporter by using the binaries for Mac ([btptf_0.1.0-beta1_darwin_amd64](https://github.com/SAP/terraform-exporter-btp/releases/download/v0.1.0-beta1/btptf_0.1.0-beta1_darwin_amd64)) or Windows ([btptf_0.1.0-beta1_windows_amd64.exe](https://github.com/SAP/terraform-exporter-btp/releases/download/v0.1.0-beta1/btptf_0.1.0-beta1_windows_amd64.exe)) the following error occurs:
    
    ´´´´
     read doc failed for data-sources, btp_subaccount: could not find docs for data-sources btp_subaccount
    ´´´´

### Expected Behavior

_No response_

### Steps To Reproduce

_No response_

### User's Role Collections

_No response_

### Add screenshots to help explain your problem

_No response_

### Additional context

_No response_

### LINKED ISSUE #185: [BUG]Exporter will not work in machines without Go installed
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What version of the Terraform exporter are you using?

0.1.0-beta

### What version of the Terraform CLI are you using?

1.8.0

### What type of issue are you facing

bug report

### Describe the bug

On Systems that do not have `go` installed, the CLI will exit with error as it will not be able to execute the `go mod` command:

```
🚀 Creation of resource list started ...

⣾  collecting resources
2024/11/18 19:15:08 read doc failed for data-sources, btp_subaccount: could not find docs for data-sources btp_subaccount
```
As this error is not handled the propagated error is the read doc failed error.

### Expected Behavior

The binary ideally should not need go installed as a prerequisite.

### Steps To Reproduce

_No response_

### User's Role Collections

_No response_

### Add screenshots to help explain your problem

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by ANUGRAHG**
## Purpose
- Removed the Go dependency to run the exporter CLI. The exporter now utilizes the [go-git](https://github.com/go-git/go-git) to clone resource documents from GitHub. Closes #185 
- Fixed read doc failed error. Closes #182 
 
## Does this introduce a breaking change?

```
[ ] Yes
[x] No
```

## Pull Request Type

What kind of change does this Pull Request introduce?

```
[x] Bugfix
[ ] Feature
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test

- Test the code via automated test

```bash
make test
```

- Additional test steps

```
...
```

## What to Check

Verify that the following are valid:

- Automated tests are executed successfully

- ...

## Checklist for reviewer

The following organizational tasks must be completed before merging this PR:

- [x] The PR is assigned to the Project board and a status is set (typically "in review").
- [x] The PR has the matching labels assigned to it.
- [x] The PR has a milestone assigned to it.
- [x] If the PR closes an issue, the issue is referenced.
- [x] Possible follow-up issues are created and linked.

### CODE REVIEW

File: pkg/tfutils/tfDocs.go [hunk 1]
```diff
@@ -59,35 +61,20 @@ func getRepositoryPath(githubHost, organization, provider, version string) (stri
...
-		return "", fmt.Errorf("error running 'go mod download -json' command in %q dir for module: %w\n\nOutput: %s", currentWd, err, output)
-	}
-
-	target := struct {
-		Version string
-		Dir     string
-		Error   string
-	}{}
+	gitRepoDownloadPath := TmpFolder + "/" + relativePath
+	gitUrl := "https://" + gitPath
```
Comment (vipinvkmenon) [inline pkg/tfutils/tfDocs.go:66]: we can use the same `relativePath`

File: pkg/tfutils/tfDocs.go [hunk 2]
```diff
@@ -51,6 +51,8 @@ var repoPaths sync.Map
 
 func getRepositoryPath(githubHost, organization, provider, version string) (string, error) {
 	relativePath := fmt.Sprintf("%s/%s/terraform-provider-%s", githubHost, organization, provider)
+	gitPath := relativePath
```
[the remark this answers is not in the document]
Comment (ANUGRAHG) [reply pkg/tfutils/tfDocs.go]: Removed the version from the relativePath and then directly using it for giturl.

File: pkg/tfutils/tfDocs.go (hunk 1 above)
Comment (ANUGRAHG) [reply pkg/tfutils/tfDocs.go]: done

File: pkg/tfutils/tfDocs.go [hunk 3]
```diff
@@ -59,35 +61,20 @@ func getRepositoryPath(githubHost, organization, provider, version string) (stri
...
-		Dir     string
-		Error   string
-	}{}
+	gitRepoDownloadPath := TmpFolder + "/" + relativePath
+	gitUrl := "https://" + gitPath
 
-	if err := json.Unmarshal(output, &target); err != nil {
-		return "", fmt.Errorf("error parsing output of 'go mod download -json' for module: %w", err)
-	}
+	_, er := git.PlainClone(gitRepoDownloadPath, false, &git.CloneOptions{
```
[the remark this answers is not in the document]
Comment (ANUGRAHG) [reply pkg/tfutils/tfDocs.go]: done

**Review by vipinvkmenon — APPROVED** [review]

**Review by vipinvkmenon — APPROVED** [review]

**Pull request merged**